test/threads.jl

```diff
@@ -141,3 +141,91 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic memory ordering
+type CommBuf{T}
+    var1::Ref{T}
```

Comment:
I want to ensure that the atomic variables live in different cache lines, thus I allocate them as arrays with 100 elements. I use `Ref` to access them conveniently. Otherwise, I could access `var1[50]`, or write a small accessor function.